packages/web-client/app/utils/web3-strategies/layer2-chain.ts

```diff
@@ -102,16 +104,13 @@ export default abstract class Layer2ChainWeb3Strategy
         console.error('Error in display_uri callback', err);
         return;
       }
+      // if we get here when a user loads a page, then it means that the user did not have
+      // a connection from local storage. We can safely say they are initialized
+      this.isInitializing = false;
```

Comment:
Because `provider.enable` doesn't resolve until a user connects their wallet, `this.isInitializing = false` under `provider.enable` will never run if the user is not connected, and they will never see the workflow. This catches the case where a user does not need to reconnect.